### PR TITLE
Added the Troubleshooting Guide for the MindsDB Inference Handler

### PR DESCRIPTION
**Pull request by MinuraPunchihewa**
## Description

This PR adds the troubleshooting guide for the MindsDB Inference handler as part of the steps for making it enterprise-ready.

## Type of change

- [x] 📄 This change is a documentation update

## Verification Process

To ensure the changes are working as expected:

 - [ ]   Test Location: Specify the URL or path for testing.
 - [ ]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

## Checklist:

- [X] My code follows the style guidelines(PEP 8) of MindsDB.
- [X] I have appropriately commented on my code, especially in complex areas.
- [X] Necessary documentation updates are either made or tracked in issues.
- [ ] Relevant unit and integration tests are updated or added - N/A

### CODE REVIEW

File: docs/integrations/ai-engines/mindsdb_inference.mdx [hunk 1]
```diff
@@ -71,3 +71,27 @@ Here is the output:
...
 For an overview of the models supported, visit the [following docs](https://docs.mdb.ai/). This list will help you quickly identify the right models for your needs.
+
+## Troubleshooting Guide
+
+<Warning>
+`Authentication Error`
+
+* **Symptoms**: Failure to authenticate to MindsDB Inference Endpoints.
+* **Checklist**:
+    1. Make sure the your MindsDB Inference Endpoints account is active.
```
Comment (martyna-mindsdb) [inline docs/integrations/ai-engines/mindsdb_inference.mdx:82]: A small typo in this line - `Make sure **that** your...`

File: mindsdb/integrations/handlers/mindsdb_inference/README.md [hunk 2]
```diff
@@ -71,3 +71,27 @@ Here is the output:
...
 For an overview of the models supported, visit the [following docs](https://docs.mdb.ai/). This list will help you quickly identify the right models for your needs.
+
+## Troubleshooting Guide
+
+<Warning>
+`Authentication Error`
+
+* **Symptoms**: Failure to authenticate to MindsDB Inference Endpoints.
+* **Checklist**:
+    1. Make sure the your MindsDB Inference Endpoints account is active.
```
Comment (martyna-mindsdb) [inline mindsdb/integrations/handlers/mindsdb_inference/README.md:82]: As above:
A small typo in this line - `Make sure **that** your...`

**Comment by MinuraPunchihewa** [comment]
@martyna-mindsdb I have fixed the typos. By the way, can you think of anything else that needs to be included here, Martyna?

**Comment by martyna-mindsdb** [comment]
@MinuraPunchihewa 
In fact, I haven't encountered any issue with MindsDB Inference, except for the authentication error, which you already listed.

As for the `SQL queries failing...` issue, I'm not sure if it is relevant here, because the way we use the models via MindsDB Inference is by sending a message/prompt to the model (not a query). Am I missing something?

**Comment by MinuraPunchihewa** [comment]
> @MinuraPunchihewa In fact, I haven't encountered any issues with MindsDB Inference, except for the authentication error, which you already listed.
> 
> As for the `SQL queries failing...` issue, I'm not sure if it is relevant here, because the way we use the models via MindsDB Inference is by sending a message/prompt to the model (not a query). Am I missing something?

Yes, you are right. The queries that can be run are quite similar to the OpenAI handler actually. So, it is possible to run queries like this:
```
SELECT output.sentiment, input.review
FROM example_db.demo_data.amazon_reviews AS input
JOIN mindsdb.sentiment_classifier AS output
LIMIT 3;
```
Running such queries can potentially lead to such errors, but the context is a little different. Shall I add different examples here?

**Comment by martyna-mindsdb** [comment]
@MinuraPunchihewa 
That's right. One thing to break down - there are two ways of using MindsDB Inference:

1. Calling MindsDB Inference endpoints via Python SDK. This can result in the `authentication error`. There are no SQL queries being run here.
2. Connecting MindsDB Inference as an integration to MindsDB. Then, one can run SQL queries that may result in the second error listed.

Does that make sense? I think we can separate these two usage examples of MindsDB Inference and mention that in Troubleshooting.

**Comment by MinuraPunchihewa** [comment]
> @MinuraPunchihewa That's right. One thing to break down - there are two ways of using MindsDB Inference:
> 
> 1. Calling MindsDB Inference endpoints via Python SDK. This can result in the `authentication error`. But, there are no SQL queries being run here.
> 2. Connecting MindsDB Inference as an integration to MindsDB. Then, one can run SQL queries that may result in the second error listed.
> 
> Does that make sense? I think we can separate these two usage examples of MindsDB Inference and mention that in Troubleshooting.

@martyna-mindsdb Do you think we should add the use of it via the Python SDK in this section of the docs? Since this is the documentation for integrations, is it relevant here?

**Review by martyna-mindsdb — APPROVED** [review]

**Comment by martyna-mindsdb** [comment]
@MinuraPunchihewa 
This looks good now. Thank you.

**Review by ZoranPandovski — APPROVED** [review]

**Pull request merged**